frame/transaction-payment/rpc/runtime-api/src/lib.rs

```diff
@@ -31,4 +31,16 @@ sp_api::decl_runtime_apis! {
 		fn query_info(uxt: Block::Extrinsic, len: u32) -> RuntimeDispatchInfo<Balance>;
 		fn query_fee_details(uxt: Block::Extrinsic, len: u32) -> FeeDetails<Balance>;
 	}
+
+	pub trait TransactionPaymentCallApi<Balance, Call>
+	where
+		Balance: Codec + MaybeDisplay,
+		Call: Codec,
+	{
+		/// Query information of a dispatch class, weight, and fee of a given encoded `Call`.
```

Comment:
removed panic, since the implementation itself is not supposed to panic, only upstream state_call